srtcore/core.cpp

```diff
@@ -7596,10 +7657,12 @@ int CUDT::processData(CUnit* unit)
    else if ((packet.m_iSeqNo & PUMASK_SEQNO_PROBE) == 1)
       m_RcvTimeWindow.probe2Arrival(pktsz);
 
-   m_ullTraceBytesRecv += pktsz;
-   m_ullBytesRecvTotal += pktsz;
-   ++ m_llTraceRecv;
-   ++ m_llRecvTotal;
+   CGuard::enterCS(m_StatsLock);
+   m_stats.traceBytesRecv += pktsz;
+   m_stats.bytesRecvTotal += pktsz;
+   ++ m_stats.traceRecv;
+   ++ m_stats.recvTotal;
+   CGuard::leaveCS(m_StatsLock);
```

Comment:
The most frequently locked block of the receiver. Profiler: ~0,3% of total CPU counts.